app/classes/query/modules/lookup_names.rb

```diff
@@ -97,59 +97,91 @@ def add_other_spellings(min_names)
         ids = min_names.map { |min_name| min_name[1] || min_name[0] }
         return [] if ids.empty?
 
-        Name.connection.select_rows(%(
-          SELECT #{minimal_name_columns} FROM names
-          WHERE COALESCE(correct_spelling_id, id) IN (#{clean_id_set(ids)})
-        ))
+        Name.
+          where((Name[:correct_spelling_id].coalesce(Name[:id])).
+                in(clean_id_set(ids).split(",").map(&:to_i))).
+          pluck(*minimal_name_columns)
       end
 
       def add_synonyms(min_names)
         ids = min_names.map { |min_name| min_name[2] }.reject(&:nil?)
         return min_names if ids.empty?
 
         min_names.reject { |min_name| min_name[2] } +
-          Name.connection.select_rows(%(
-            SELECT #{minimal_name_columns} FROM names
-            WHERE synonym_id IN (#{clean_id_set(ids)})
-          ))
+          Name.where(synonym_id: clean_id_set(ids).split(",")).
+          pluck(*minimal_name_columns)
       end
 
       def add_subtaxa(min_names)
```

Comment:
I probably got too carried away with `add_subtaxa` and `add_subtaxa`.
After just fixing the SQL injection stuff, I noticed that
- each of these methods was sometimes hitting the database twice and 
- that could be improved by chaining the AR/Arel methods,
- and that I could avoid intermediate assignments to `min_names`,
- and when all of that was done, I ended up with ABC offenses, which I fixed by extracting methods.